nextflow_schema.json

```
@@ -629,6 +629,88 @@
+                    "fa_icon": "fas fa-angle-double-right"
+                },
+                "mapping_mapad_gapreadenddistance": {
+                    "type": "number",
+                    "default": 5,
+                    "description": "Specify the --gap_dist_ends parameter for mapAD, which defines the distance from the read ends in which no gaps are permitted.",
+                    "help_text": "Configures mapAD's `--gap_dist_ends` parameter which defines the distance from the read ends in which no gaps are permitted.\n\n> Modifies mapAD map parameter: `--gap_dist_ends`",
+                    "fa_icon": "fas fa-step-forward"
+                },
+                "mapping_mapad_maxnumopengaps": {
+                    "type": "number",
+                    "default": 2,
+                    "description": "Specify the number of gaps allowed for alignment with mapAD.",
+                    "help_text": "Specify the number of gaps allowed for mapping with mapAD.\n\n> Modifies mapAD map parameter: `--max_num_gaps_open`",
+                    "fa_icon": "fas fa-people-arrows"
+                },
+                "mapping_mapad_baseerrorrate": {
+                    "type": "number",
+                    "default": 0.02,
+                    "description": "Specify the base error rate parameter for mapAD.",
+                    "help_text": "Configures mapAD's base error rate parameter. The `-D` parameter adjusts the base error rate which is one of the underlying parts of mapAD's scoring model. Should not normally need to be adjusted.\n\n> Modifies mapAD map parameter: `-D`",
+                    "fa_icon": "fas fa-percentage"
+                },
+                "mapping_mapad_ignorebasequal": {
+                    "type": "boolean",
+                    "description": "Set the --ignore_base_quality flag, which instructs mapAD to ignore base quality values in its scoring model.",
+                    "help_text": "Without this flag, mapAD is more permissive in terms of mismatches if the quality of the base is low.\n\n> Modifies mapAD map parameter: `--ignore_base_quality`",
+                    "fa_icon": "fas fa-asterisk"
+                },
+                "mapping_mapad_stacklimitabort": {
+                    "type": "boolean",
+                    "description": "Set the --stack_limit_abort flag, which instructs mapAD to not try and recover from a full backtracking stack.",
+                    "help_text": "Without this flag, mapAD tries to recover from a full backtracking stack by removing low-scoring sub-alignments. This can help to recover from difficult to map reads, at the cost of a slower alignment. Setting this flag will speed up the alignment process.\n\n> Modifies mapAD map parameter: `--stack_limit_abort`",
```

I don't really follow what exactly the behaviour does here, and I think the 'backtracking stack' jargon might not be help either.

```suggestion
                    "help_text": "By default, mapAD will remove low-scoring sub-alignments, helping to recover from difficult to map reads, at the cost of a slower alignment. Setting this flag will speed up the alignment process by not removing these.\n\n> Modifies mapAD map parameter: `--stack_limit_abort`",
```

Is that what you mean?

---

Right! I'm going to flip the default for this in mapAD and reword this accordingly..